src/Compilers/CSharp/Portable/Declarations/MergedNamespaceDeclaration.cs

```diff
@@ -186,5 +186,8 @@ public new ImmutableArray<MergedNamespaceOrTypeDeclaration> Children
                 return _lazyChildren;
             }
         }
+
+        ImmutableArray<INamespaceOrTypeDeclaration> INamespaceDeclaration.Children
+            => this.Children.SelectAsArray(d => (INamespaceOrTypeDeclaration)d);
```

Comment:
this allocates a copy.  You should just be able to return `this.Children`. you may possibly need to do `ImmutableArray.CastUp`, but tha is allocation free.